src/utils/AppDevUtils.ts

```diff
@@ -37,8 +37,26 @@ const randomCode = (length: number): string =>
     .slice(1)
     .toUpperCase();
 
+/**
+ * Validates a possible url
+ * @function
+ * @param {string} url - URL to validate
+ * @returns {boolean} true if URL is valid, false otherwise
+ * https://stackoverflow.com/questions/5717093/check-if-a-javascript-string-is-a-url/49849482
+*/
+const validateURL = (url: string): boolean => {
+  const pattern = new RegExp('^(https?:\\/\\/)?' + // protocol
+    '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' + // domain name
+    '((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
+    '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
+    '(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
+    '(\\#[-a-z\\d_]*)?$', 'i'); // fragment locator
+  return !!pattern.test(url);
```

Comment:
Double NOT operators or is this some funky regex thing? also trusting that stackoverflow link for that pattern 😬

Reply:
double bang is actually something I have been reading more about after encountering it in a couple of code bases: https://airbnb.io/javascript/#comparison--unneeded-ternary
it's a nice way to indicate that the _only_ possible return values are `true` and `false` (instead of some truthy or falsy value).

also have been playing around with that regex pattern for a decent number of tests and it seems to be pretty all-encompassing... hopefully it stays that way, lol...

Reply:
updated to use URL class - much cleaner